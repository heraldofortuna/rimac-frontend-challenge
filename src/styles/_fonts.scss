@font-face {
  font-family: 'BR Sonoma Regular';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Regular'), url('/fonts/BRSonoma-Regular-BF654c45266c042.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Regular Italic';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Regular Italic'), url('/fonts/BRSonoma-RegularItalic-BF654c452681c11.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Thin';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Thin'), url('/fonts/BRSonoma-Thin-BF654c45255ffe0.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Thin Italic';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Thin Italic'), url('/fonts/BRSonoma-ThinItalic-BF654c45268d3f9.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma ExtraLight';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma ExtraLight'), url('/fonts/BRSonoma-ExtraLight-BF654c45265af8d.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma ExtraLight Italic';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma ExtraLight Italic'), url('/fonts/BRSonoma-ExtraLightItalic-BF654c4525a5046.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Light';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Light'), url('/fonts/BRSonoma-Light-BF654c452608e0f.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Light Italic';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Light Italic'), url('/fonts/BRSonoma-LightItalic-BF654c45266aa83.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Medium';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Medium'), url('/fonts/BRSonoma-Medium-BF654c45266edd1.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Medium Italic';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Medium Italic'), url('/fonts/BRSonoma-MediumItalic-BF654c45267d45f.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma SemiBold';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma SemiBold'), url('/fonts/BRSonoma-SemiBold-BF654c45268c340.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma SemiBold Italic';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma SemiBold Italic'), url('/fonts/BRSonoma-SemiBoldItalic-BF654c452696350.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Bold';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Bold'), url('/fonts/BRSonoma-Bold-BF654c4526823f5.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Bold Italic';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Bold Italic'), url('/fonts/BRSonoma-BoldItalic-BF654c4525c9c27.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Black';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Black'), url('/fonts/BRSonoma-Black-BF654c4525506bf.woff') format('woff');
}
  

@font-face {
  font-family: 'BR Sonoma Black Italic';
  font-style: normal;
  font-weight: normal;
  src: local('BR Sonoma Black Italic'), url('/fonts/BRSonoma-BlackItalic-BF654c45268988e.woff') format('woff');
}