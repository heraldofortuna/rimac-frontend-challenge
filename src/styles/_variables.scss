$grey-100-color: #03050F;
$grey-80-color: #2B304E;
$grey-30-color: #CCD1EE;
$grey-10-color: #F8F9FF;
$blue-berry-600-color: #4F4FFF;
$neutrals-700-color: #141938;
$neutrals-600-color: #7981B2;
$neutrals-0-color: #FFFFFF;

$sonoma-font: 'BR Sonoma Regular', sans-serif;
$lato-font: 'Lato', sans-serif;

$breakpoint-mobile: 480px;
$breakpoint-desktop: 768px;