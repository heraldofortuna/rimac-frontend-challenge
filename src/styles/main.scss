@use 'fonts';
@import url('https://fonts.googleapis.com/css2?family=Lato:wght@300;400;700;900&display=swap');

* {
  padding: 0;
  margin: 0;
}

body {
  background-color: $grey-10-color;
  font-family: $sonoma-font;
  color: $grey-100-color;
}

#root {
  min-height: 100vh;
  display: flex;
  flex-direction: column;
}