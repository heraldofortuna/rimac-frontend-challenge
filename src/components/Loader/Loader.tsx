import styles from "./Loader.module.scss";

const Loader: React.FC = () => {
  return (
    <div className={styles.loaderContainer} data-testid="loader-container">
      <div className={styles.loader} data-testid="loader-element"></div>
    </div>
  );
};

export default Loader;