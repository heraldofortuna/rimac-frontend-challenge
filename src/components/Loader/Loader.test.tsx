import { render, screen } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import Loader from './Loader';
import styles from './Loader.module.scss';

describe('Loader Component', () => {
  it('renders without crashing', () => {
    render(<Loader />);
    const loaderContainer = screen.getByTestId('loader-container');
    expect(loaderContainer).toBeInTheDocument();
  });

  it('has the correct container class', () => {
    render(<Loader />);
    const loaderContainer = screen.getByTestId('loader-container');
    expect(loaderContainer).toHaveClass(styles.loaderContainer);
  });

  it('contains the loader element with correct class', () => {
    render(<Loader />);
    const loaderElement = screen.getByTestId('loader-element');
    expect(loaderElement).toHaveClass(styles.loader);
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Loader />);
    expect(asFragment()).toMatchSnapshot();
  });
});