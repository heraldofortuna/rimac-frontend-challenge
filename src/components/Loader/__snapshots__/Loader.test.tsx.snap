// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Loader Component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="_loaderContainer_0c7d59"
    data-testid="loader-container"
  >
    <div
      class="_loader_0c7d59"
      data-testid="loader-element"
    />
  </div>
</DocumentFragment>
`;
